spec/support/stub_article_service.rb

```diff
@@ -0,0 +1,59 @@
+##
+# Module included for RSpec tests to stub the article search service
+module StubArticleService
+  def stub_article_service(type: :multiple, docs:)
+    raise 'Article search service stubbed without any documents.' if docs.blank?
+
+    allow_any_instance_of(ArticleController).to receive(:setup_eds_session).and_return('abc123')
+    case type
+    when :multiple
+      expect_any_instance_of(Eds::SearchService).to receive(:search_results).and_return(
+        [StubAticleResponse.new(docs), nil]
+      )
+    when :single
+      raise "Single document response requsted but #{docs.length} provided." if docs.many?
+      expect_any_instance_of(Eds::SearchService).to receive(:fetch).and_return(
+        [StubAticleResponse.new(docs.first), docs.first]
+      )
+    else
+      raise "Unknown article stub type #{type} provided."
+    end
+  end
+
+  class StubAticleResponse < Blacklight::Solr::Response
```

Comment:
Typo here: `StubAticleResponse` should be `StubArticleResponse`? You have it spelled that way in various places.